doc/articles/Surfaces.md

```diff
@@ -0,0 +1,256 @@
+
+Surfaces are used to continuously present color texture data to users in an OS Window, HTML `<canvas>`, or other similar outputs.
+The `webgpu.h` concept of @ref WGPUSurface is similar to WebGPU's [`GPUCanvasContext`](https://gpuweb.github.io/gpuweb/#canvas-context) but includes additional options to control behavior or query options that are specific to the environment the application runs in. In other GPU APIs, similar concepts are called "default framebuffer", "swapchain" or "presentable".
+
+To use a surface, there is a one-time setup, then additional per-frame operations.
+The one time setup is: environment-specific creation (to wrap a `<canvas>`, `HWND`, `Window`, etc.), querying capabilities of the surface, and configuring the surface.
+Per-frame operations are: getting a current @ref WGPUSurfaceTexture to render content to, rendering content, presenting the surface, and when appropriate reconfiguring the surface (typically when the window is resized).
+
+Sections below give more details about these operations, including the specification of their behavior.
+
+## Surface Creation {#Surface-Creation}
+
+A @ref WGPUSurface is child object of a @ref WGPUInstance and created using `::wgpuInstanceCreateSurface`.
+The description of a @ref WGPUSurface is a @ref WGPUSurfaceDescriptor with a sub-descriptor chained containing the environment-specific objects used to identify the surface.
+
+Surfaces that can be presented to using webgpu.h (but not necessarily by all implementations) are:
+
+ - `ANativeWindow` on Android with @ref WGPUSurfaceSourceAndroidNativeWindow
+ - `CAMetalLayer` on various Apple OSes like macOS and iOS with @ref WGPUSurfaceSourceMetalLayer
+ - `<canvas>` HTML elements in Emscripten (targeting WebAssembly) with `WGPUSurfaceSourceCanvasHTMLSelector_Emscripten`
+ - `HWND` on Windows with @ref WGPUSurfaceSourceWindowsHWND
+ - `Window` using Xlib with @ref WGPUSurfaceSourceXlibWindow
+ - `wl_surface` on Wayland systems with @ref WGPUSurfaceSourceWaylandSurface
+ - `xcb_window_t` using XCB windows with @ref WGPUSurfaceSourceXCBWindow
+
+For example, creating an @ref WGPUSurface from an `HWND` is done like so:
+
+```c
+WGPUSurfaceSourceWindowsHWND hwndDesc = {
+    .chain = { .sType = WGPUSType_SurfaceSourceWindowsHWND, },
+    .hinstance = GetModuleHandle(nullptr),
+    .hwnd = myHWND,
+};
+WGPUSurfaceDescriptor desc {
+    .nextInChain = &hwndDesc.chain,
+    .label = "Main window",
+};
+WGPUSurface surface = wgpuInstanceCreateSurface(myInstance, &desc);
+```
+
+In addition, @ref WGPUSurfaces have a bunch of internal fields that could be represented like this (in C/Rust-like pseudocode):
+
+```cpp
+struct WGPUSurface {
+    // The parent object
+    WGPUInstance instance;
+
+    // The current configuration
+    Option<WGPUSurfaceConfiguration> config = None;
+
+    // The frame, if any.
+    Option<WGPUTexture> currentFrame = None;
+};
+```
+
+The behavior of `::wgpuInstanceCreateSurface``(instance, descriptor)` is:
```

Comment:
Is this "``"  on purpose? If not, should it be?
```Suggestion
The behavior of `::wgpuInstanceCreateSurface(instance, descriptor)` is:
```

Note that there are several other instances.

Reply:
That's on purpose, otherwise Doxygen doesn't know to put a link to `wgpuInstanceCreateSurface`. This file isn't meant to be read directly, but to be processed by Doxygen.